### ISSUE TITLE
Cluster can't recover/replicate at Pod startup

### ISSUE THREAD

**Issue by Pflegusch** [bitnami/bitnami-docker-mariadb-galera#59]
**Description**
We are using the [Helm Chart from Bitnami](https://github.com/bitnami/charts/tree/master/bitnami/mariadb-galera) in **production** and we are facing some issues at the moment concerning pod restarts. The current image version is docker.io/bitnami/mariadb-galera:10.5.11-debian-10-r7 for the MariaDB image and docker.io/bitnami/mysqld-exporter:0.13.0-debian-10-r27 for the mysqld exporter. We were running a 3 replica stateful set of the Chart and had no problems until one of the pods restarted and failed to recover. We had to scale down to 1 and currently are only running 1 node. If we scale up, we can see that in the logs of the only running pod there gets a file created named ``mariabackup.backup.log`` under ``/bitnami/mariadb/data`` in the instance we scale up the stateful set. The content is the following:
```
[00] 2021-08-02 08:36:17 Connecting to MySQL server host: localhost, user: `mariabackup, password: set, port: 3306, socket: /opt/bitnami/mariadb/tmp/mysql.sock
[00] 2021-08-02 08:36:17 Failed to connect to MySQL server: Access denied for user '`mariabackup'@'localhost' (using password: YES).
```
We assume that this is the reason why the other pods can't restart anymore. Why is there suddenly this character at the username specified? At the same kubernetes cluster, we have 2 additional 3 replica cluster running where the problem doesn't persist and the content of the corresponding file also seems to be good:
```
[00] 2021-07-29 12:07:40 Connecting to MySQL server host: localhost, user: mariabackup, password: set, port: 3306, socket: /opt/bitnami/mariadb/tmp/mysql.sock
[00] 2021-07-29 12:07:40 Using server version 10.5.11-MariaDB-log
........
``` 
I looked up the my.cnf and the environment variables at the pod, they all have the correct ``mariabackup`` in there. 

**Steps to reproduce the issue:**
Don't know, seems to happen randomly?

**Describe the results you received:**
Cannot scale up the replica set anymore 

**Describe the results you expected:**
After scaling to 1 and then back up to 3, the nodes should join the first node and build up a 3-node cluster.

**Comment by marcosbc** [comment]
Hi @Pflegusch, this seems like a really weird issue. Did this only happen once? Could you also share the version of the MariaDB Galera chart that you are using?

It could be a deployment misconfiguration, so I'd recommend you to look into the environment variables set in the running container (via the `env` command) in case you can see something strange, and also the running processes (via the `ps aux` command). If you can find a "`mariabackup" there, then it's likely related to a deployment misconfiguration.

**Comment by Pflegusch** [comment]
Hi @marcosbc ,

the chart version is ``mariadb-galera-5.10.3`` with app version ``10.5.11``. I checked the deployment configurations multiple times, I cant find a problem there. The thing is that I deployed the other clusters with the same values.yaml as this broken one, just with different passwords and persistent volume sizes, so I dont think the problem lies there. I can rescale those clusters without problems too. I also checked the environment variables within the running container and even there it states ``MARIADB_GALERA_MARIABACKUP_USER=mariabackup``. I also tried reading the ``wsrep_sst_auth`` variable, however it is obfuscated and I cant read it. 

The next approach would be to update the image itself and restart the cluster in hope to fix the problem, however since we are running in production I would rather understand the problem first before doing unnecessary changes, maybe you have something I can try to fix it. 

Can you tell me what script/workflow gets started if a cluster actually starts joining? Maybe there is some script which is broken or something, idk

**Comment by Pflegusch** [comment]
P.S: Also checked some scripts now:
```
I have no name!@mariadb-mariadb-galera-0:/opt/bitnami/scripts$ grep -rnw . -e 'mariabackup'
./libmariadbgalera.sh:319:wsrep_sst_method=mariabackup
./libmariadbgalera.sh:393:# Ensure the mariabackup user exists for host 'localhost' and has full access (galera)
./libmariadbgalera.sh:397:#   $1 - mariabackup user
./libmariadbgalera.sh:406:    debug "Configure mariabackup user credentials"
./mariadb-env.sh:194:export MARIADB_GALERA_DEFAULT_SST_METHOD="mariabackup"
./mariadb-env.sh:196:export MARIADB_GALERA_DEFAULT_MARIABACKUP_USER="mariabackup"
```

**Comment by Pflegusch** [comment]
I think I found the missconfiguration: In ``/opt/bitnami/mariadb/conf/my.cnf``, it states wsrep_sst_auth=\`mariabackup:ourpw\`, while on the other clusters that isnt the case.

**Comment by marcosbc** [comment]
Hi @Pflegusch, currently the container image creates the configuration file via a heredoc template:

```
...
wsrep_sst_auth=${DB_GALERA_DEFAULT_MARIABACKUP_USER}:${DB_GALERA_DEFAULT_MARIABACKUP_PASSWORD}
```

Those environment variables come from `MARIADB_GALERA_DEFAULT_MARIABACKUP_USER` and `MARIADB_GALERA_DEFAULT_MARIABACKUP_PASSWORD` which have `mariabackup` and empty values, respectively (from `mariadb-env.sh`).

However, in the chart, the story is different because the configuration file is getting mounted from the `values.yaml`: https://github.com/bitnami/charts/blob/master/bitnami/mariadb-galera/values.yaml#L402

```
wsrep_sst_auth="root:"
```

And since it is getting mounted by default, I don't understand how you are seeing a different value for `wsrep_sst_auth` in the configuration file. Could it be possible that you modified the default configuration?

**Comment by Pflegusch** [comment]
Hi @marcosbc, thank you for the reply. I looked into it for some hours again and I run out of ideas. I don't think I modified the default configuration. What I've tried so far apart from the written above:

- Running ``grep -rnws /opt -e '`mariabackup'`` gave me a match at ``/opt/bitnami/mariadb/conf/my.cnf``, changing that such that the special character is gone doesn't fix the issue. If I delete the pod such that it restarts, the old value is there again even tho the file is not persistent (maybe changing that could be a solution? don't think so tho)
- Checking all configmaps, secrets and values files doesn't fix the issue as there is no problem with them
- Scaling down to zero, upgrading the image in the values file and then restarting the cluster doesn't work, even across multiple nodes

Just don't understand where that character comes from, it isnt on the other clusters and the environment variables you described above also don't show it.

Next step would be to setup a cluster from scratch and importing a current backup from the broken one, maybe that will help
